### PR TITLE
tests: shellchecks part 3

### PR DESCRIPTION
**Pull request by bboozzoo**
Part 3

### CODE REVIEW

File: tests/main/interfaces-alsa/task.yaml [hunk 1]
```diff
@@ -18,11 +18,12 @@ details: |
 
 prepare: |
     echo "Given a snap declaring a plug on the alsa interface is installed"
+    #shellcheck source=tests/lib/snaps.sh
     . $TESTSLIB/snaps.sh
     install_generic_consumer alsa
 
 restore: |
-    rm -f *.error /dev/snd/mysnd-dev
+    rm -f ./*.error /dev/snd/mysnd-dev
```
Comment (zyga) [inline tests/main/interfaces-alsa/task.yaml]: I'm curious to know what was the warning here. Do you remember?

File: tests/main/interfaces-bluetooth-control/task.yaml [hunk 2]
```diff
@@ -23,35 +25,37 @@ execute: |
...
     echo "And the snap is able to read class"
     cat /sys/class/bluetooth/*/name | tee class
-    [ $(su -l -c "/snap/bin/generic-consumer.cmd cat /sys/class/bluetooth/*/name" test) = $(cat class) ]
+    [ "$(su -l -c '/snap/bin/generic-consumer.cmd cat /sys/class/bluetooth/*/name' test)" = "$(cat class)" ]
 
     echo "And the snap is able to read dev"
-    cat $BTDEV/*/power/control | tee control
-    [ $(su -l -c "/snap/bin/generic-consumer.cmd cat $BTDEV/*/power/control" test) = $(cat control) ]
+    cat "$BTDEV"/*/power/control | tee control
+    #shellcheck disable=SC2016
```
Comment (zyga) [inline tests/main/interfaces-bluetooth-control/task.yaml]: Why are we disabling this? Is BTDEV exported? Could we use the trick recommended by the wiki to expand it here?

```
[ "$(su -l -c '/snap/bin/generic-consumer.cmd cat '"$BTDEV"'/*/power/control' test)' = "$(cat control)" ]
```

File: tests/main/interfaces-content-mimic/task.yaml [hunk 3]
```diff
@@ -21,11 +22,17 @@ execute: |
     # entries that should be correctly replicated inside the mimic.
     
     check_existing_files() {
+        #shellcheck disable=SC2016
```
Comment (zyga) [inline tests/main/interfaces-content-mimic/task.yaml]: This is all correct, it's too bad there's no way to express this in a shellcheck happy way, apart from disabling this check.

File: tests/main/interfaces-alsa/task.yaml (hunk 1 above)
Comment (bboozzoo) [reply tests/main/interfaces-alsa/task.yaml]:
```
In - line 1:
rm -f *.error /dev/snd/mysnd-dev
      ^-- SC2035: Use ./*glob* or -- *glob* so names with dashes won't become options.
```

**Comment by chipaca** [comment]
some of the fixes are buggy :-)

File: tests/main/interfaces-browser-support/task.yaml [hunk 4]
```diff
@@ -39,24 +39,25 @@ prepare: |
...
-    rm -f *.error /var/tmp/test
+    rm -f ./*.error /var/tmp/test
     for file in $OWNED_FILES $READABLE_FILES $READABLE_WITH_SANDBOX_FILES; do
-        rm -f $file
+        rm -f "$file"
     done
 
-    for dir in $(cat created_dirs); do
-        rm -rf $dir
+    while read -r dir < created_dirs; do
```
Comment (chipaca) [inline tests/main/interfaces-browser-support/task.yaml]: you want to create infinite loops? 'cause this is how you create infinite loops

File: tests/main/interfaces-browser-support/task.yaml [hunk 5]
```diff
@@ -44,9 +44,9 @@ restore: |
         rm -f "$file"
     done
 
-    while read -r dir < created_dirs; do
+    while read -r dir; do
```
Comment (bboozzoo) [inline tests/main/interfaces-browser-support/task.yaml]: Ayy, silly me. Thanks for fixing this.

File: tests/main/install-sideload/task.yaml [hunk 6]
```diff
@@ -20,50 +20,51 @@ restore: |
...
     . $TESTSLIB/dirs.sh
 
     echo "Sideload desktop snap"
     snap install --dangerous ./basic-desktop_1.0_all.snap
-    expected="\[Desktop Entry\]\n\
-    Name=Echo\n\
-    Comment=It echos stuff\n\
-    Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/basic-desktop_echo.desktop $SNAP_MOUNT_DIR/bin/basic-desktop.echo\n"
-    cat /var/lib/snapd/desktop/applications/basic-desktop_echo.desktop | grep -Pzq "$expected"
+    diff -u <(head -n4 /var/lib/snapd/desktop/applications/basic-desktop_echo.desktop) - <<-EOF
```
Comment (mvo5) [inline tests/main/install-sideload/task.yaml]: nice!

File: tests/main/interfaces-avahi-observe/task.yaml [hunk 7]
```diff
@@ -20,20 +22,23 @@ prepare: |
...
-    rm -f *.error
+    rm -f ./*.error
+    #shellcheck source=tests/lib/pkgdb.sh
     . $TESTSLIB/pkgdb.sh
     distro_purge_package avahi-daemon
     distro_auto_remove_packages
 
 execute: |
-    avahi_dbus_call="dbus-send --system --print-reply --dest=org.freedesktop.Avahi / org.freedesktop.Avahi.Server.GetHostName"
+    avahi_dbus_call() {
```
Comment (mvo5) [inline tests/main/interfaces-avahi-observe/task.yaml]: nice!

File: tests/main/interfaces-bluetooth-control/task.yaml (hunk 2 above)
Comment (bboozzoo) [reply tests/main/interfaces-bluetooth-control/task.yaml]: Done

**Pull request merged**